catalog/catalog.go

```diff
@@ -185,3 +197,33 @@ func TableNameFromIdent(ident table.Identifier) string {
 func NamespaceFromIdent(ident table.Identifier) table.Identifier {
 	return ident[:len(ident)-1]
 }
+
+func getMetadataPath(locationPath string, newVersion int) (string, error) {
+	if newVersion < 0 {
+		return "", fmt.Errorf("invalid table version: %d must be a non-negative integer", newVersion)
+	}
+
+	metaDataPath, err := url.JoinPath(strings.TrimLeft(locationPath, "/"), "metadata", fmt.Sprintf("%05d-%s.metadata.json", newVersion, uuid.New().String()))
+	if err != nil {
+		return "", fmt.Errorf("failed to build metadata path: %w", err)
+	}
+
+	return metaDataPath, nil
+}
```

Comment:
are these specific to glue? 

And wouldn't generating the uuid and version etc. be something that should be in the table package rather than in here?

Reply:
Yeah I think there are a few couple of things like this that came out of me "using it in anger", I was interested in your thoughts but wasn't super stressed as it is private.

Hopefully we can push some of this down to the table.